src/ImageRunModal.jsx

```diff
@@ -632,6 +587,79 @@
         return owner === systemOwner;
     };
 
+    isFormValid = (validationFailed) => {
+        return Object.keys(validationFailed).length == 0;
+    };
+
+    async validateContainerName(containerName) {
+        try {
+            await client.containerExists(this.isSystem(), containerName);
+        } catch (error) {
+            return;
+        }
+        return _("Name already in use");
+    }
+
+    async validateForm() {
+        const { publish, volumes, env, containerName } = this.state;
+        const validationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            let publishPortValidation;
+            publishPortValidation = validatePublishPort(a, "IP", publishPortValidation);
+            publishPortValidation = validatePublishPort(a, "hostPort", publishPortValidation);
+            publishPortValidation = validatePublishPort(a, "containerPort", publishPortValidation);
+
+            return publishPortValidation;
+        });
+        if (Object.keys(publishValidation).length > 0)
+            validationFailed.publish = publishValidation;
+
+        const volumesValidation = volumes.map(a => {
+            let volumesValidation;
+            volumesValidation = validateVolume(a, "hostPath", volumesValidation);
+            volumesValidation = validateVolume(a, "containerPath", volumesValidation);
+
+            return volumesValidation;
+        });
+        if (Object.keys(volumesValidation).length > 0)
+            validationFailed.volumes = volumesValidation;
+
+        const envValidation = env.map(a => {
+            let envValidation;
+            envValidation = validateEnvVar(a, "envKey", envValidation);
+            envValidation = validateEnvVar(a, "envValue", envValidation);
+
+            return envValidation;
+        });
+        if (Object.keys(envValidation).length > 0)
+            validationFailed.env = envValidation;
+
+        validationFailed.containerName = await this.validateContainerName(containerName);
+
+        this.setState({ validationFailed });
+
+        return this.isFormValid(validationFailed);
+    }
+
+    /* Updates a validation object of the whole dynamic list's form (e.g. the whole port-mapping form)
+    *
+    * Arguments
+    *   - key: [publish/volumes/env] - Specifies the validation of which dynamic form of the Image run dialog is being updated
+    *   - value: An array of validation errors of the form. Each item of the array represents a row of the dynamic list.
+    *            Index needs to corellate with a row number
+    */
+    dynamicListOnValidationChange(value, key) {
+        const validationFailedDelta = { ...this.state.dialogValue.validationFailed };
+
+        validationFailedDelta[key] = value;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230907-093411-a9f37bd4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)